### ISSUE TITLE
Paginator returns the same cursor if use paginationKeys: ['createdAt']

### ISSUE THREAD

**Issue by alprtv** [benjamin658/typeorm-cursor-pagination#14]
Hi, thanks for lib)
It's work fine, but when I try to paginate by "createdAt" date paginator (after the second try) always returns the same cursor.

take=1

Steps:
 1. 
```
"id": "238c07a1-ef71-4f2c-928b-49e214b4185f"
"cursor": "Y3JlYXRlZEF0OjE1OTYzNDk4NTA1Mjk="
```
2.
```
"id": "de750b6b-959a-4a54-a591-f47fb1b512d9"
"cursor": "Y3JlYXRlZEF0OjE1OTYzNDk2MzI3MjY="
```
3.
```
"id": "de750b6b-959a-4a54-a591-f47fb1b512d9",
"cursor": "Y3JlYXRlZEF0OjE1OTYzNDk2MzI3MjY="
```
<details>
  <summary>in Entity</summary>
  
  ```typescript
  @CreateDateColumn({ type: 'timestamp' })
  createdAt: Date;
  ```
  
</details>

<details>
  <summary>in DB</summary>

```
[
   {
      "id":"238c07a1-ef71-4f2c-928b-49e214b4185f",
      "createdAt":"2020-08-02T06:30:50.529Z"
   },
   {
      "id":"de750b6b-959a-4a54-a591-f47fb1b512d9",
      "createdAt":"2020-08-02T06:27:12.726Z"
   },
   {
      "id":"e6ee30ed-386e-477d-930c-b7505fce9a56",
      "createdAt":"2020-08-02T06:26:07.508Z"
   },
   {
      "id":"f432f4d9-0293-4cf1-b298-cd3c62adc7fd",
      "createdAt":"2020-08-02T06:24:36.519Z"
   },
...
]
```
  
</details>

<details>
  <summary>My code</summary>
  
  ```typescript
const queryBuilder = this.productOrdersRepository
      .createQueryBuilder(ProductOrder.tableName)
      .where(`${ProductOrder.tableName}."venueId" = :venueId`, {
        venueId: query.venueId,
      });

    let paginator: Paginator<ProductOrder>;

    if (query.cursor) {
      paginator = buildPaginator({
        alias: ProductOrder.tableName,
        entity: ProductOrder,
        paginationKeys: ['createdAt'],
        query: {
          limit: query.take,
          afterCursor: query.cursor,
          order: 'DESC',
        },
      });
    } else {
      paginator = buildPaginator({
        alias: ProductOrder.tableName,
        entity: ProductOrder,
        paginationKeys: ['createdAt'],
        query: {
          limit: query.take,
          order: 'DESC',
        },
      });
    }

    const { data, cursor } = await paginator.paginate(queryBuilder);
```

</details>

lib version
`"typeorm-cursor-pagination": "^0.4.1",`

**Comment by benjamin658** [comment]
Hi @alprtv , thanks for reporting the issue.

Please make sure that the `createdAt` is a unique column.

**Comment by alprtv** [comment]
Thx for answer)

> Please make sure that the `createdAt` is a unique column.

in my bd this field is not unique :( 
but `createdAt` date has never been repeated in the table.
do i understand correctly, for pagination, the field must be unique?
```
 @CreateDateColumn({ type: 'timestamp', unique: true })
  createdAt: Date;
```

**Comment by bjoveski** [comment]
I experienced something similar, but I believe it has to do with the precision of the date columns. postgres is storing timestampts with 6 digits of precision (microseconds), but javascript supports only 3. This means that a notification N with 12:00:00.000001 as createdAt in the db, will have 12:00:00.000 as createdAt in the javascript land. Since the pagination works by asking for the more-recent records than the last record, N will again be included because 12:00:00.000001 > 12:00:00.000.

You can fix this by using `@CreateDateColumn({ precision: 3 })`.


In addition, you don't necessarily need to enforce uniqueness of the `createdAt` column - instead you can pass `paginationKeys: ['createdAt', 'id'],` which will always be unique

**Comment by benjamin658** [comment]
I think @bjoveski is right, thanks for your answer.
